js/node/src/session_options_helper.cc

```diff
@@ -321,44 +321,46 @@ void ParseSessionOptions(const Napi::Object options, Ort::SessionOptions& sessio
   // external data
   if (options.Has("externalData")) {
     auto externalDataValue = options.Get("externalData");
-    ORT_NAPI_THROW_TYPEERROR_IF(!externalDataValue.IsArray(), options.Env(),
-                                "Invalid argument: sessionOptions.externalData must be an array.");
-    auto externalData = externalDataValue.As<Napi::Array>();
-    std::vector<std::basic_string<ORTCHAR_T>> paths;
-    std::vector<char*> buffs;
-    std::vector<size_t> sizes;
+    if (!externalDataValue.IsNull() && !externalDataValue.IsUndefined()) {
```

Comment:
Do we need to produce an error or warning if the externalData is specified and there is no data?

Reply:
in N-API, it's different between "the sessionOptions object does not have 'externalData' property" and "sessionOptions.externalData is `undefined`". We usually don't make a difference for this situation.

If user specify session options like this:
```js
const sessionOptions = {
  // .. other options

  externalData: undefined
};
```

This should still work. The old Node.js binding will fail in this case.

Reply:
In other words, I think it is fine if externalData is null or undefined. In this case it is considered there is no external data.